pkg/volume/aws_ebs/aws_util.go

```diff
@@ -190,6 +194,21 @@ func getDiskByIdPaths(partition string, devicePath string) []string {
 		}
 	}
 
+	// We need to find NVME volumes, which are mounted on a "random" nvme path ("/dev/nvme0n1"),
+	// and we have to get the volume id from the nvme interface
+	awsVolumeID, err := volumeID.MapToAWSVolumeID()
+	if err != nil {
+		glog.Warningf("error mapping volume %q to AWS volume: %v", volumeID, err)
+	} else {
+		nvmeName := strings.Replace(string(awsVolumeID), "-", "", -1)
+		nvmePath, err := findNvmeVolume(nvmeName)
```

Comment:
This might be a daft idea but can we rely on `ls /dev/disk/by-id/vme-Amazon_Elastic_Block_Store_*` to tell us device path rather than having to rely on using ioctl's at all?

For example:

```
lrwxrwxrwx 1 root root 13 Nov 27 15:56 /dev/disk/by-id/nvme-Amazon_Elastic_Block_Store_vol0dssdsdsd -> ../../nvme0n1
lrwxrwxrwx 1 root root 15 Nov 27 15:56 /dev/disk/by-id/nvme-Amazon_Elastic_Block_Store_vol0dsdsds-part1 -> ../../nvme0n1p1
lrwxrwxrwx 1 root root 13 Nov 27 16:07 /dev/disk/by-id/nvme-Amazon_Elastic_Block_Store_vol0dsdsdsds -> ../../nvme1n1
```

Reply:
Great suggestion - trying it in #56607 (so we don't wipe the comments here, but hopefully most of them become moot as that code is much simpler!)